include/srf/experimental/modules/segment_modules.hpp

```diff
@@ -0,0 +1,171 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2021-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "nlohmann/json.hpp"
+
+#include "srf/segment/object.hpp"
+
+#include <map>
+#include <string>
+#include <typeinfo>
+#include <utility>
+#include <vector>
+
+namespace srf::segment {
+class Builder;
+}
+
+namespace srf::modules {
+
+class SegmentModule
+{
+  public:
+    using segment_module_port_map_t     = std::map<std::string, std::shared_ptr<segment::ObjectProperties>>;
+    using segment_module_port_t         = std::shared_ptr<segment::ObjectProperties>;
+    using segment_module_typeinfo_map_t = std::map<std::string, const std::type_info*>;
```

Comment:
Not sure I understand the use of `const std::type_info*`.

Can you explain this choice vs. `std::type_index`?

Reply:
I think this is just more of the old style way to do it, and I find it somewhat more intuitive for one reason:

If you have something like `std::map<std::string, std::type_index> some_map`, you can't return `some_map["key"], because if the key doesn't exist map will try to default construct one and type_index doesn't have a default constructor.

Based on this, you end up having to return the equivalent of `some_map.find("key")->first`.

End of the day its an implementation detail that I'm not terribly attached to, other than I do think its slightly more intuitive.